<a href="https://colab.research.google.com/github/swapnilzambare/EDA-Airbnb-Data-Analysis-Capstone-Project/blob/main/EDA_Airbnb_data_analysis_capstone_project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> Since 2008, guests and hosts have used Airbnb to expand on traveling possibilities and present a more unique, personalized way of experiencing the world. Today, Airbnb became one of a kind service that is used and recognized by the whole world. Data analysis on millions of listings provided through Airbnb is a crucial factor for the company. These millions of listings generate a lot of data - data that can be analyzed and used for security, business decisions, understanding of customers' and providers' (hosts) behavior and performance on the platform, guiding marketing initiatives, implementation of innovative additional services and much more. </b>

## <b>This dataset has around 49,000 observations in it with 16 columns and it is a mix between categorical and numeric values. </b>

## <b> Explore and analyze the data to discover key understandings (not limited to these) such as : 
* What can we learn about different hosts and areas?
* What can we learn from predictions? (ex: locations, prices, reviews, etc)
* Which hosts are the busiest and why?
* Is there any noticeable difference of traffic among different areas and what could be the reason for it? </b>

# **1. Acquire and loading data**

# **Load python libraries**

In [10]:
# Python libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import matplotlib.image as mpimg
%matplotlib inline
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

# **Load dataset**

In [2]:
# Mounting drive for import csv file data
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [11]:
# Importing csv file of Airbnb NYC 2019 from drive
airbnb= pd.read_csv("/content/drive/MyDrive/Copy of Airbnb NYC 2019.csv")

In [12]:
# Check file
airbnb

id  ... availability_365
0          2539  ...              365
1          2595  ...              355
2          3647  ...              365
3          3831  ...              194
4          5022  ...                0
...         ...  ...              ...
48890  36484665  ...                9
48891  36485057  ...               36
48892  36485431  ...               27
48893  36485609  ...                2
48894  36487245  ...               23

[48895 rows x 16 columns]

# **Understanding data**
**After we load the dataset, we need to understand the dataset by using various techniques. First, we need to look for information on how big is our dataset. By using shape attributes, we get to know our data size from a number of rows which consist of listing index, and the number of columns with the content of every features related to the index.**

In [13]:
# Shape of airbnb data which shows number of rows and columns
airbnb.shape

(48895, 16)

**Then we check all the data type of every column if it already matches our requirement. For instance, we need a numerical data type (integer and float) on the longitude and latitude, for listing names we need to make sure the data is using string/object data type.**

In [6]:
# Data type
airbnb.dtypes

id                                  int64
name                               object
host_id                             int64
host_name                          object
neighbourhood_group                object
neighbourhood                      object
latitude                          float64
longitude                         float64
room_type                          object
price                               int64
minimum_nights                      int64
number_of_reviews                   int64
last_review                        object
reviews_per_month                 float64
calculated_host_listings_count      int64
availability_365                    int64
dtype: object

**We found out that our dataset has 48895 listings in 16 catogorical data. The features include listing name, host id, location information, location coordinate, room type, the price per night, and so on.**

In [14]:
# Checking info non- null and data type
airbnb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

In [15]:
# Descriptive statistics summary
airbnb.describe()

id  ...  availability_365
count  4.889500e+04  ...      48895.000000
mean   1.901714e+07  ...        112.781327
std    1.098311e+07  ...        131.622289
min    2.539000e+03  ...          0.000000
25%    9.471945e+06  ...          0.000000
50%    1.967728e+07  ...         45.000000
75%    2.915218e+07  ...        227.000000
max    3.648724e+07  ...        365.000000

[8 rows x 10 columns]

**Next, we look up all the unique values of the ‘neighbourhood_group’ that is consists of a list of all the New York region.**
**From list below we undertand that New York is divided in 5 neighbourhood group**.

In [16]:
# Neighbourhood group in New York
ngg=airbnb['neighbourhood_group'].unique()
print('Neighbourhood Group =',list(ngg))

Neighbourhood Group = ['Brooklyn', 'Manhattan', 'Queens', 'Staten Island', 'Bronx']


In [17]:
# Neighbourhood group are further divided in neighbourhood 
# Checking total neighbourhood and print them
ng=airbnb['neighbourhood'].unique()
print('Total neighbourhood =',len(ng))
print('Neighbourhood =',list(ng))

Total neighbourhood = 221
Neighbourhood = ['Kensington', 'Midtown', 'Harlem', 'Clinton Hill', 'East Harlem', 'Murray Hill', 'Bedford-Stuyvesant', "Hell's Kitchen", 'Upper West Side', 'Chinatown', 'South Slope', 'West Village', 'Williamsburg', 'Fort Greene', 'Chelsea', 'Crown Heights', 'Park Slope', 'Windsor Terrace', 'Inwood', 'East Village', 'Greenpoint', 'Bushwick', 'Flatbush', 'Lower East Side', 'Prospect-Lefferts Gardens', 'Long Island City', 'Kips Bay', 'SoHo', 'Upper East Side', 'Prospect Heights', 'Washington Heights', 'Woodside', 'Brooklyn Heights', 'Carroll Gardens', 'Gowanus', 'Flatlands', 'Cobble Hill', 'Flushing', 'Boerum Hill', 'Sunnyside', 'DUMBO', 'St. George', 'Highbridge', 'Financial District', 'Ridgewood', 'Morningside Heights', 'Jamaica', 'Middle Village', 'NoHo', 'Ditmars Steinway', 'Flatiron District', 'Roosevelt Island', 'Greenwich Village', 'Little Italy', 'East Flatbush', 'Tompkinsville', 'Astoria', 'Clason Point', 'Eastchester', 'Kingsbridge', 'Two Bridges', 'Q

**The region area is divided further by the Airbnb into 221 areas called 'neighbourhood' areas for booking purposes. We will use the ‘neighbourhood’ columns to look at which region that has the Airbnb listing.**

In [18]:
# Room type that Airbnb provides
rt=airbnb['room_type'].unique()
print('Room Type =',list(rt))

Room Type = ['Private room', 'Entire home/apt', 'Shared room']


**From the list above, we see that Airbnb have room type. Based on the information on the Airbnb website, the definition of each room type are:**
## *   **Private room** ✈
**Guests have exclusive access to the bedroom/sleeping area of the listing. Other parts area such as the living room, kitchen, and bathroom are likely open either to the host even to other guests**.
## *   **Entire home/apt** ✈
**Guests have the whole place for themselves. It usually includes a bedroom, bathroom, and kitchen**.
## *   **Shared Room** ✈
**Guest sleep in a bedroom or a common area that could be shared with others. bold text**

# **2. Cleaning dataset**

# **Checking column with missing values**

In [19]:
# Let’s check first if there are any missing values within our dataset
airbnb.isnull().sum()

id                                    0
name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64

# **Removing redundant variables**
**In our case, the missing values that are observed do not need too much treatment. Looking into our dataset, we can state columns ‘id' and ‘host_name’, ‘last_review’ are irrelevant and unethical for further data exploration analysis. Therefore, we can get rid of those columns.**

In [20]:
# Removing redundant variable
airbnb.drop(['id','host_name','last_review'],axis=1,inplace=True)
airbnb.head()

name  ...  availability_365
0                Clean & quiet apt home by the park  ...               365
1                             Skylit Midtown Castle  ...               355
2               THE VILLAGE OF HARLEM....NEW YORK !  ...               365
3                   Cozy Entire Floor of Brownstone  ...               194
4  Entire Apt: Spacious Studio/Loft by central park  ...                 0

[5 rows x 13 columns]

# **Replacing all the missing values**
**Next, we need to replace all the missing values in the ‘review_per_month’ and 'price' column with 0 (zero) to make sure the missing values do not interfere with our analysis**

In [22]:
# Fill NAN values with zero
airbnb['reviews_per_month'].fillna(0,inplace=True)

In [23]:
# Fill NAN values with zero
airbnb['price'].fillna(0,inplace=True)